client/web/compose/src/views/Admin/Modules/Records/List.vue

```diff
@@ -102,6 +102,41 @@ export default {
     },
   },
 
+  watch: {
+    module (module) {
+      const { meta = { ui: {} } } = module || {}
+
+      let fields = ((meta.ui || {}).admin || {}).fields || []
+      fields = fields.length ? fields : module.fields
+
+      // Init block
+      const block = new compose.PageBlockRecordList({
+        blockIndex: 0,
+        options: {
+          moduleID: module.moduleID,
```

Comment:
You dont need to reinit the whole block, it has a watcher on options, so just updating the moduleID should be enough
Also can you turn in into the regular watcher structure (the handler ()), also you need to check if module exists